### PR TITLE
Feature/sc 6588/app home page that supports multiple care

### PR DESCRIPTION
**Pull request by lvancraen**
Story: https://app.shortcut.com/capable/story/6588/app-home-page-that-supports-multiple-care-plans

Adds support for multiple care plans. There was a decent amount of refactorization work needs to make the code cleaner and allow for the multiple plans. I've also changed some of the names of the components to be more directly inline with what the component actually does.

https://user-images.githubusercontent.com/15220513/188004012-423f134e-23de-4aeb-8dab-60519a245c69.mov

### CODE REVIEW

File: src/pages/Home.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
+import { useState } from "react";
+import { useParams } from "react-router-dom";
+import { Container, Typography, Box, CardMedia, Avatar, Skeleton, Stack } from "@mui/material";
 
+import { AboutCard, ArrowLink, ControlledTabs, GoalCards, TaskCards } from "../components";
+import { useCarePlans, useCurrentPatient } from "../fetchDataHooks";
 import gravatar from "../utils/gravatar";
 
-function Header() {
+function Header({ carePlan, carePlans }) {
```
Comment (lawyu89) [inline src/pages/Home.js:9]: doest look like you need the carePlans param.

File: src/pages/Home.js [hunk 2]
```diff
@@ -50,17 +69,82 @@ function Header() {
...
+    currentCarePlan = openCarePlans[0];
+  } else if (completedCarePlans.length > 0) {
+    currentCarePlan = completedCarePlans[0];
+  }
+
   return (
     <>
-      <Header />
-      <TabsWithContent />
+      <Header carePlan={currentCarePlan} carePlans={allCarePlans} />
```
Comment (lawyu89) [inline src/pages/Home.js:146]: if you dont need carePlans param:
```suggestion
      <Header carePlan={currentCarePlan} />
```

File: src/fetchDataHooks/fetcher.js [hunk 3]
```diff
@@ -11,6 +11,7 @@ import api from "../capableApi/index";
 // );
 const fetcher = (resource, action, args) => {
   if (args) {
+    console.log("Calling api");
```
Comment (lawyu89) [inline src/fetchDataHooks/fetcher.js:14]: cruft?

**Review by lawyu89 — APPROVED** [review]
looks really good to me 💯

**Review by ezekielp — APPROVED** [review]
🙌

File: src/pages/CarePlans.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  }
+
+  if (isErrorCarePlans || isErrorGoalsByStatus) {
+    return (
+      <div>Woops something went wrong...</div>
+    );
+  }
+
+  const openCarePlans = carePlans.filter(carePlan => carePlan.status === "active");
+  const OpenCarePlans = openCarePlans.length > 0
```
[the remark this answers is not in the document]
Comment (achidu) [reply src/pages/CarePlans.js:78]: Agreed — that's my mistake in the designs 😞 Could we please change the {active} Care Plan section title on the list view to "Active" @lvancraen?

File: src/fetchDataHooks/fetcher.js (hunk 3 above)
Comment (lvancraen) [reply src/fetchDataHooks/fetcher.js:14]: Good catch! I missed that, thanks

**Review by ezekielp — APPROVED** [review]

**Pull request merged**